run_task.py

```diff
@@ -146,14 +151,37 @@ def run(self):
 					finish_msg = "| Task finished with return code " + str(process.poll()) + " |"
 					finish_msg = "\n" + ("-" * len(finish_msg)) + "\n" + finish_msg + "\n" + ("-" * len(finish_msg))
 					output_panel.write(finish_msg)
+					if self.has_file:
+						self.window.focus_view(self.window.active_view())
 					break
 				if out_line:
 					output_panel.write(out_line)
+
 		else:
-			try:
-				subprocess.Popen(self.args, cwd=self.cwd)
-			except Exception as ex:
-				sublime.error_message(ErrorMessage.task_execution_failed(self.task_name, str(ex)))
+			if not self.has_file:
+				try:
+					subprocess.Popen(self.args, cwd=self.cwd)
+				except Exception as ex:
+					sublime.error_message(ErrorMessage.task_execution_failed(self.task_name, str(ex)))
+			else:
+				try:
+					si = subprocess.STARTUPINFO()
+					si.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+					si.wShowWindow = subprocess.SW_HIDE # default
```

Comment:
As per the [docs](https://docs.python.org/3/library/subprocess.html#subprocess.STARTUPINFO), the `STARTUPINFO` class is only available on Windows, so this code will crash on Linux and macOS. At the same time, I believe the `si.wShowWindow = subprocess.SW_HIDE` is needed to prevent the command shell window from opening when running the task on Windows, but please correct me if I'm wrong.

To fix the above, and address my previous comments, I would suggest to rewrite the `run()` method as follows:
```python
if platform.system() == 'Windows':
	si = subprocess.STARTUPINFO()
	si.dwFlags |= subprocess.STARTF_USESHOWWINDOW
	si.wShowWindow = subprocess.SW_HIDE
else:
	si = None

if self.show_output_panel:
	output_panel = OutputPanel(self.window)
	output_panel.show()
	try:
		process = subprocess.Popen(self.args, bufsize=1, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, cwd=self.cwd, universal_newlines=True, startupinfo=si)
	except Exception as ex:
		sublime.error_message(ErrorMessage.task_execution_failed(self.task_name, str(ex)))
		return
	while True:
		out_line = process.stdout.readline()
		if out_line == '' and process.poll() is not None:
			finish_msg = "| Task finished with return code " + str(process.poll()) + " |"
			finish_msg = "\n" + ("-" * len(finish_msg)) + "\n" + finish_msg + "\n" + ("-" * len(finish_msg))
			output_panel.write(finish_msg)
			break
		if out_line:
			output_panel.write(out_line)
else:
	try:
		subprocess.Popen(self.args, cwd=self.cwd, startupinfo=si)
	except Exception as ex:
		sublime.error_message(ErrorMessage.task_execution_failed(self.task_name, str(ex)))
```
, but it needs to be tested on Windows.